pkg/lib/catalogsource/image_template.go

```diff
@@ -0,0 +1,356 @@
+package catalogsource
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/blang/semver/v4"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/util/jsonpath"
+)
+
+const (
+
+	// regex capture group names
+
+	// cap_grp_kubemajorv is a capture group name for a kube major version
+	cap_grp_kubemajorv = "kubemajorv"
```

Comment:
nit: `snake_case` is not conventional in go

Reply:
agreed... see https://github.com/operator-framework/operator-lifecycle-manager/pull/2281#discussion_r677535154

Reply:
fixed